drivers/wifi/esp/esp.c

```diff
@@ -722,6 +722,21 @@ static int esp_mgmt_ap_disable(const struct device *dev)
 	return ret;
 }
 
+static void esp_configure_hostname(struct esp_data *data)
+{
+#if defined(CONFIG_NET_HOSTNAME_ENABLE)
+	char cmd[64];
+
+	snprintk(cmd, sizeof(cmd), "AT+CWHOSTNAME=\"%s\"",
+		 net_hostname_get());
+	cmd[sizeof(cmd) - 1] = '\0';
+
+	modem_cmd_send(&data->mctx.iface, &data->mctx.cmd_handler,
+		       NULL, 0, cmd, &data->sem_response,
+		       ESP_CMD_TIMEOUT);
+#endif
```

Comment:
probably does not make much difference but we could add here
```
#else 
     ARG_UNUSED(data);
#endif 
```

Reply:
Is this related to MISRA? I tried to grep for `ARG_UNUSED` and didn't found anything in Zephyr's `doc` directory. So I would like to understand some background instead of blindly following this advise.

Reply:
Yes, this was related to some MISRA stuff. We have this macro in many places of the code base.